src/explorer/webview.ts

```diff
@@ -137,59 +147,46 @@ class ExplorerWebview {
-            for (let i = 0; i < codeFlowTrees.length; i++) {
-                codeFlowTrees.item(i).removeEventListener("click", this.onCodeFlowTreeClickedBind);
+            const codeFlowTrees: HTMLCollectionOf<HTMLUListElement> = getDocumentElementsByClassName(document, "codeflowtreeroot", HTMLUListElement);
+            for (const codeFlowTree of codeFlowTrees) {
+                codeFlowTree.removeEventListener("click", this.onCodeFlowTreeClicked.bind(this));
             }
 
-            let element = document.getElementById("expandallcodeflow");
-            if (element !== null) {
-                element.removeEventListener("click", this.onExpandAllClickedBind);
-            }
+            let element: HTMLDivElement = getDocumentElementById(document, "expandallcodeflow", HTMLDivElement);
+            element.removeEventListener("click", this.onCollapseAllClicked.bind(this));
 
-            element = document.getElementById("collapseallcodeflow");
-            if (element !== null) {
-                element.removeEventListener("click", this.onCollapseAllClickedBind);
-            }
+            element = getDocumentElementById(document, "collapseallcodeflow", HTMLDivElement);
+            element.removeEventListener("click", this.onCollapseAllClicked.bind(this));
 
-            element = document.getElementById("codeflowverbosity");
-            if (element !== null) {
-                element.removeEventListener("click", this.onVerbosityChangeBind);
-            }
+            const verbosityElement: HTMLInputElement = getDocumentElementById(document, "codeflowverbosity", HTMLInputElement);
+            verbosityElement.removeEventListener("click", this.onVerbosityChange.bind(this));
         }
 
         if (this.hasAttachments) {
-            const attachmentTrees = document.getElementsByClassName("attachmentstreeroot");
-            for (let i = 0; i < attachmentTrees.length; i++) {
-                attachmentTrees.item(i).removeEventListener("click", this.onAttachmentClicked);
+            const attachmentTrees: HTMLCollectionOf<HTMLUListElement> = getDocumentElementsByClassName(document, "attachmentstreeroot", HTMLUListElement);
+            for (const attachmentTree of attachmentTrees) {
+                attachmentTree.removeEventListener("click", this.onAttachmentClicked.bind(this));
             }
         }
 
         // Clear Result Details
-        const header = document.getElementById("resultdetailsheader");
-        while (header.children.length > 0) {
-            header.removeChild(header.children.item(0));
-        }
-
-        const container = document.getElementById("resultdetailscontainer");
-        while (container.children.length > 0) {
-            container.removeChild(container.children.item(0));
-        }
+        removeElementChildren(document.getElementById("resultdetailsheader"));
+        removeElementChildren(document.getElementById("resultdetailscontainer"));
```

Comment:
I'm unclear on the purpose of `removeElementChildren()` over the simpler `document.getElementById("resultdetailscontainer").innerHTML = ""`?

Reply:
With struct null checks turned on, 
document.getElementById("resultdetailscontainer").innerHTML = "".

is not valid as getElementById can return null.

And optional property access
document.getElementById("resultdetailscontainer")?.innerHTML = ""

isn't allowed.

So... the best you can do is have removeElementChildren helper do what you suggest and have it take Element | null.



---
In reply to: [397701937](https://github.com/microsoft/sarif-vscode-extension/pull/211#discussion_r397701937) [](ancestors = 397701937)